### PR TITLE
style(report): Add a Markdown format template

### PR DESCRIPTION
**Pull request by ohmer**
## Description

See https://github.com/aquasecurity/trivy/issues/3201

## Related issues
- Close https://github.com/aquasecurity/trivy/issues/3201

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Review by DmitriyLewen — COMMENTED** [review]
Hello @ohmer 
Thanks for your work!

I added some comments. Take a look please.

Also i have 1 more question:
What about secrets?
It would be great to add secrets to the template as well.

File: contrib/markdown.tpl [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        <th>Installed Version</th>
+        <th>Fixed Version</th>
+    </tr>
+    {{- range .Vulnerabilities }}
+    <tr>
+        <td><code>{{ escapeXML .PkgName }}</code></td>
+        <td>{{ escapeXML .VulnerabilityID }}</td>
+        <td>{{ escapeXML .Severity }}</td>
+        <td>{{ escapeXML .InstalledVersion }}</td>
+        <td>{{ escapeXML .FixedVersion }}</td>
```
[the remark this answers is not in the document]
Comment (ohmer) [reply contrib/markdown.tpl:22]: Hi @DmitriyLewen,

I considered this and preferred a lighter output. With the fact that GitHub markdown renderer automatically links to a CVE information page, detailed information are really a click away. Example from private repo:

image

Just a matter of personnal taste, I don't have a strong opinion about it.

**Comment by ohmer** [comment]
> Hello @ohmer Thanks for your work!
> 
> I added some comments. Take a look please.
> 
> Also i have 1 more question: What about secrets? It would be great to add secrets to the template as well.

Thanks for you review @DmitriyLewen, let me get back to you on the secret. I have not covered this case as I am not implemented secrets scanning for $WORK yet.

File: contrib/markdown.tpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (DmitriyLewen) [reply contrib/markdown.tpl:22]: Strange... this doesn't work for my test `*.md` file - https://github.com/DmitriyLewen/testproject/blob/main/test.md
Also users can use this template not only for GitHub.
I think would be better to add `Title` field.

**Comment by DmitriyLewen** [comment]
Adding secrets is possible for you, i understood you correct?

And 1 more request: can you add test for this template? We have tests for some templates [here](https://github.com/aquasecurity/trivy/blob/34d505ad1497dd11c1865efa05ec96032d8fedae/integration/client_server_test.go#L314-L340).

**Comment by 64J0** [comment]
Up. This feature would be very cool!

**Comment by ohmer** [comment]
> Hi @ohmer.
> 
> Do you intend to keep working on/finish this PR? If not, I would like to work on it. 

$JOB made me real busy, sorry. Thanks for reminding me. I have made a little progress on adding secret. Feel free to offer a second PR, that's open source with community contributions welcomed after all.

**Comment by itaysk** [comment]
Apologies for the late response. We are trying to control the built-in output formatters to just the basic ones that the core team can maintain (like table, json, etc), while at the same time provide extensible mechanism for community to create and share template without needing to push it upstream. This is already possible today, as this PR shows - someone created a gotemplate and solved their problem. The only thing missing is a way for trivy users discover these templates - If you want, you can share your work in the "show and tell" GitHub discussion, or even open a PR to add it to the Trivy documentation under the "ecosystem" section. We also have plans to improve the discovery and distribution of templates in the future. For now, we have to close this PR as it is very specific for some scenario and not something we intend to maintain in the core repo.

**Pull request closed without merging**